core/arch/arm/plat-hikey/spi_test.c

```diff
@@ -48,45 +72,57 @@
  */
 void spi_test(void)
 {
-	struct pl061_data platform_pl061_data;
+	struct pl061_data platform_pl061_data __maybe_unused;
 	struct pl022_data platform_pl022_data;
-	vaddr_t gpio6_base = nsec_periph_base(GPIO6_BASE);
+	vaddr_t __maybe_unused gpio6_base = nsec_periph_base(GPIO6_BASE);
 	vaddr_t spi_base = nsec_periph_base(SPI_BASE);
 	uint8_t tx[3] = {0x01, 0x80, 0x00};
 	uint8_t rx[3] = {0};
 	size_t i, j, len = 3;
+	bool local_cs = false;
```

Comment:
Ah, now I see the reason for `__maybe_unused`. Make this `local_cs` a parameter to the test function instead. Then `spi_test()` could call `_spi_test(false);`, and possibly also `_spi_test(true);` to test both modes?